server/routes/punishment_numberOfAdditionalDays/numberOfAdditionalDaysValidation.ts

```diff
@@ -0,0 +1,27 @@
+import { FormError } from '../../@types/template'
+
+type NumberOfAddedDaysForm = {
+  days: number
+}
+
+const errors: { [key: string]: FormError } = {
+  MISSING_DAYS: {
+    href: '#days',
+    text: 'Enter how many days the punishment will last',
+  },
+  DAYS_TOO_FEW: {
+    href: '#days',
+    text: 'Enter one or more days',
+  },
+  NOT_NUMERICAL: {
+    href: '#days',
+    text: 'Enter a number of days',
+  },
+}
+
+export default function validateForm({ days }: NumberOfAddedDaysForm): FormError | null {
+  if (Number.isNaN(days) || typeof days === 'string') return errors.NOT_NUMERICAL
+  if (Number.isInteger(days) && days <= 0) return errors.DAYS_TOO_FEW
+  if (days === undefined || days === null || !days) return errors.MISSING_DAYS
```

Comment:
Minor readability point - you could have these `if`s in the same order as the collection of `errors`